firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/persistence/CrashlyticsReportPersistence.java

```diff
@@ -270,6 +234,61 @@ private void capFinalizedReports() {
         getAllFilesInDirectory(priorityReportsDirectory), getAllFilesInDirectory(reportsDirectory));
   }
 
+  private File getSessionDirectoryById(String sessionId) {
+    return new File(openSessionsDirectory, sessionId);
+  }
+
+  private void synthesizeReportFile(File sessionDirectory, long sessionEndTime) {
+    final List<File> eventFiles =
+        getFilesInDirectory(sessionDirectory, (f, name) -> name.startsWith(EVENT_FILE_NAME_PREFIX));
+
+    // Only process the session if it has associated events
+    if (eventFiles.isEmpty()) {
+      return;
+    }
+
+    // TODO: Handle all nullable return values
+
+    Collections.sort(eventFiles);
+    final List<Event> events = new ArrayList<>();
+    boolean isHighPriorityReport = false;
+    for (File eventFile : eventFiles) {
+      final String eventJson = readTextFile(eventFile);
+      if (eventJson == null) {
+        // TODO: Handle null case
+        continue;
+      }
+      final Event event = TRANSFORM.eventFromJson(eventJson);
+      isHighPriorityReport = isHighPriorityReport || isHighPriorityEventFile(eventFile.getName());
```

Comment:
```suggestion
      isHighPriorityReport |= isHighPriorityEventFile(eventFile.getName());
```

Reply:
Personally I find this less clear, and would prefer we keep it as is if it's all the same to you.